commons-rest/filters/src/main/java/org/eclipse/kapua/commons/rest/filters/CORSResponseFilter.java

```diff
@@ -97,34 +97,49 @@ public void destroy() {
-            KapuaId scopeId = KapuaSecurityUtils.getSession() != null ? KapuaSecurityUtils.getSession().getScopeId() : null;
-
-            if (checkOrigin(origin, scopeId)) {
-                // Origin matches at least one defined Endpoint
-                httpResponse.addHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
-                httpResponse.addHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
-                httpResponse.addHeader("Vary", HttpHeaders.ORIGIN);
+
+        // Depending on the type of the request the KapuaSession might not yet be present.
+        // For preflight request or session not yet established the KapuaSession will be null.
+        // For the actual request it will be available and we will check the CORS according to the scope.
+        KapuaId scopeId = KapuaSecurityUtils.getSession() != null ? KapuaSecurityUtils.getSession().getScopeId() : null;
+        String origin = StringUtils.isNotEmpty(httpRequest.getHeader(HttpHeaders.ORIGIN)) ? httpRequest.getHeader(HttpHeaders.ORIGIN) : null;
+        String fetchSite = httpRequest.getHeader(HttpHeaders.SEC_FETCH_SITE);
+
+        if (StringUtils.isEmpty(fetchSite)) {
+            logger.info("<sec-fetch-site> not provided, is the browser incompatible with this header? expect false positives for same origin cors requests filtering");
+        }
+        if (StringUtils.isEmpty(fetchSite) || //browser not compatible with sec fetch site (for example, safari), proceed with origin check anyway
+                !fetchSite.equals("same-origin")) {
+            if (origin == null) {
+                logger.info("<origin> not provided, is the browser incompatible with this header?");
             } else {
-                errorMessage = scopeId != null ?
-                        String.format("HTTP Origin not allowed: %s for scope: %s", origin, scopeId.toCompactId()) :
-                        String.format("HTTP Origin not allowed: %s", origin);
-                logger.error(errorMessage);
+                // Origin header present, so it's a CORS request. Apply all the required logics
+                httpResponse.addHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_METHODS, "GET, POST, DELETE, PUT");
+                httpResponse.addHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS, "X-Requested-With, Content-Type, Authorization");
+
+                if (checkOrigin(origin, scopeId)) {
+                    // Origin matches at least one defined Endpoint
+                    httpResponse.addHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
+                    httpResponse.addHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+                    httpResponse.addHeader("Vary", HttpHeaders.ORIGIN);
+                } else {
+                    //this log, for browsers not supporting set fetch site, logs false positive for same origin CORS. This thing is inevitable considering that here we cannot understand if the request comes from the same origin
```

Comment:
This comment I think is a bit misleading. 
This problem can occur when requests are performed by an application or via curl